controller/src/main/java/io/pravega/controller/task/Stream/StreamTransactionMetadataTasks.java

```diff
@@ -323,7 +326,7 @@ public void initializeStreamWriters(final EventStreamWriter<CommitEvent> commitW
                                                                                    final OperationContext ctx) {
         // Step 1. Validate parameters.
         CompletableFuture<Void> validate = validate(lease);
-        long maxExecutionPeriod = Math.min(MAX_EXECUTION_TIME_MULTIPLIER * lease, Duration.ofDays(1).toMillis());
+        long maxExecutionPeriod = Math.min(MAX_EXECUTION_TIME_MULTIPLIER * lease, maxTransactionExecutionTimeBound.get());
```

Comment:
`maxExecutionPeriod` represents the maximum time for which the client may extend txn lease. It will always remain 1000 * 30second = 8.33 hours...

Reply:
certain connector implementations would actually want to use a much larger lease. maxlease can be increased and the intent is to set it to upto 30 days.. and a 1000 multiplier could take it to several years.. hence we also have a maxTransactionExecutionTimebound and we take min of these two values.

Reply:
ok, the user would need to change the controller properties `Config.MAX_LEASE_VALUE` and `Config.MAX_TXN_EXECUTION_TIMEBOUND_DAYS ` to ensure his requirements are met.